images/image-alt-text-context-form.html

```diff
@@ -0,0 +1,164 @@
+                editor.ui.registry.addContextForm('alttext', {
+                    // Define the button used in our context toolbar to open the context form
+                    launch: {
+                        type: 'contextformbutton',
+                        icon: 'alt-text',
+                        tooltip: 'Image description (Alt text)'
+                    },
+
+                    // The input field label (Currently does nothing).
+                    label: 'Image description (alt text)',
+
+                    // Initial value of the context form's input field
+                    initValue: () => {
+                        // First we get the selected element. It can be either the
+                        // callout itself or any child element within it.
+                        // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.selection/#getnode
+                        const node = editor.selection.getNode();
+
+                        // Get the selected image's alt attribute text
+                        // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.domutils/#getattrib
+                        const altText = tinymce.DOM.getAttrib(node, 'alt');
+
+                        // Return the the alt text to be displayed as initial value
+                        return altText;
+                    },
+                    // Define the buttons associated with the context form, in this case
+                    // a save and cancel button
+                    commands: [
+                        {
+                            type: 'contextformbutton',
+                            primary: true,
+                            tooltip: 'Save',
+                            icon: 'checkmark',
+                            onAction: (contextFormApi) => {
+                                // Working directly with the DOM requires manually adding
+                                // the actions to the undo stack.
+                                // https://www.tiny.cloud/docs/api/tinymce/tinymce.undomanager/
+                                editor.undoManager.transact(() => {
+                                    // Get the selected element (in this case our image)
+                                    // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.selection/#getnode
+                                    const node = editor.selection.getNode();
+
+                                    // Set the alt attribute on the selected image
+                                    // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.domutils/#setattrib
+                                    const alt = tinymce.DOM.setAttrib(node, 'alt', contextFormApi.getValue());
+
+                                    // Close the context form
+                                    // https://www.tiny.cloud/docs/ui-components/contextform/#formapi
+                                    contextform.hide();
```

Comment:
Typo here, as this should be `contextFormApi` and was throwing an exception. As such, that's likely why it wasn't working in your comment above.

Reply:
Facepalm